[DOC-630][docs-beta] update PyObject to create /api/python-api/ href (#26942)
dagster-io/dagster: docs/docs-beta/CONTRIBUTING.md

There are some features in the previous docs that require changes to be made to work in the new Docusaurus-based documentation site.

### PyObject references

> To create a link to the Python API docs, the `PyObject` component is to be used. Previously, we were able to parse the Sphinx search index to determine the section that the module resides. As we no longer have this, a `section` prop was added to the component.

Reviewer: ```suggestion
To create a link to the Python API docs, use the `PyObject` component. Previously, we were able to parse the Sphinx search index to determine the section that the module resides. As we no longer have this, a `section` prop was added to the component.
```
